src/github.com/couchbase/sync_gateway/base/sharded_sequence_clock.go

```diff
@@ -0,0 +1,290 @@
+package base
+
+import (
+	"bytes"
+	"encoding/gob"
+	"expvar"
+	"fmt"
+	"sync"
+
+	"github.com/couchbase/sg-bucket"
+)
+
+var shardedClockExpvars *expvar.Map
+
+func init() {
+	shardedClockExpvars = expvar.NewMap("syncGateway_index_clocks")
+}
+
+const (
+	kIndexPartitionKey       = "_idxPartitionMap"
+	kIndexPrefix             = "_idx"
+	kCountKeyFormat          = "_idx_c:%s:count"    // key
+	kClockPartitionKeyFormat = "_idx_c:%s:clock-%d" // key, partition index
+)
+
+type PartitionStorage struct {
+	Uuid  string   `json:"uuid"`
+	Index uint16   `json:"index"`
+	VbNos []uint16 `json:"vbNos"`
+}
+
+type IndexPartitionMap map[uint16]uint16 // Maps vbuckets to index partition value
+
+type IndexPartitions struct {
+	VbMap         IndexPartitionMap
+	PartitionDefs []PartitionStorage
+}
+
+func NewIndexPartitions(partitions []PartitionStorage) *IndexPartitions {
+
+	indexPartitions := &IndexPartitions{
+		PartitionDefs: make([]PartitionStorage, len(partitions)),
+	}
+	copy(indexPartitions.PartitionDefs, partitions)
+
+	indexPartitions.VbMap = make(map[uint16]uint16)
+	for _, partition := range partitions {
+		for _, vbNo := range partition.VbNos {
+			indexPartitions.VbMap[vbNo] = partition.Index
+		}
+	}
+	return indexPartitions
+}
+
+// Index partitionsfor testing
+func TestIndexPartitions() *IndexPartitions {
```

Comment:
This is just for testing purposes?  I guess there's probably no easy way to get this code into the test suite without some significant refactoring, but I think at the very least it should be moved to the bottom of the file since it's less important for understanding the code.